src/h5web/providers/models.ts

```diff
@@ -48,6 +48,47 @@ export interface HDF5Datatype<T = HDF5Type> {
   type: T;
 }
 
+export enum MyHDF5EntityKind {
+  Group = 'group',
+  Dataset = 'dataset',
+  Datatype = 'datatype',
+  Link = 'link',
+}
+
+export interface MyHDF5Entity {
+  uid: string;
+  id?: HDF5Id;
+  name: string;
+  kind: MyHDF5EntityKind;
+  parents: MyHDF5Group[];
+  attributes: HDF5Attribute[];
+  rawLink: HDF5Link;
+  rawEntity?: HDF5Entity;
+}
+
+export interface MyHDF5Group extends MyHDF5Entity {
+  kind: MyHDF5EntityKind.Group;
+  children: MyHDF5Entity[];
+}
+
+export interface MyHDF5Dataset<
+  S extends HDF5Shape = HDF5Shape,
+  T extends HDF5Type = HDF5Type
+> extends MyHDF5Entity {
+  kind: MyHDF5EntityKind.Dataset;
+  shape: S;
+  type: T;
+}
+
+export interface MyHDF5Datatype<T = HDF5Type> extends MyHDF5Entity {
+  kind: MyHDF5EntityKind.Datatype;
+  type: T;
+}
+
+export interface MyHDF5Link extends MyHDF5Entity {
+  kind: MyHDF5EntityKind.Link;
+}
```

Comment:
This type is not used yet, but we'll most likely end up removing `rawLink` from` MyHDF5Entity` and exposing some properties of the link here.